### PR TITLE
update apple-mapkit-js types to 5.18

### PR DESCRIPTION
**Pull request by chrisdrackett**
also add docs/comments

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/wsmd/mapkit-typescript/
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by chrisdrackett** [comment]
mapkit is not on NPM, so I'm not sure how to resolve the CI failure. Looks like CI is picking up a related but also unrelated project on NPM that is not apple's mapkit (only available via a URL)

https://developer.apple.com/documentation/mapkitjs/loading_the_latest_or_a_specific_version_of_mapkit_js

**Comment by chrisdrackett** [comment]
ok, I was able to fix that issue, but now I'm getting `SyntaxError: Unexpected token } in JSON at position 82` when running the tests and I'm not sure what JSON it is even talking about :\

File: types/apple-mapkit-js/index.d.ts [hunk 1]
```diff
@@ -1,714 +1,2261 @@
-// Type definitions for MapKit JS 5.4
-// Project: https://developer.apple.com/reference/mapkitjs, https://github.com/lukemizuhashi/apple-mapkit-js
+// Type definitions for MapKit JS 5.18
```
Comment (sheetalkamat) [inline types/apple-mapkit-js/index.d.ts]: change this to `// Type definitions for non-npm package MapKit JS 5.18`

File: types/apple-mapkit-js/tslint.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
 {
-    "extends": "dtslint/dt.json"
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (sheetalkamat) [inline types/apple-mapkit-js/tslint.json]: Please remove the rules from here

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Pull request closed without merging**